phpBB/phpbb/search/fulltext_sphinx.php

```diff
@@ -430,22 +435,27 @@ protected function config_generate()
 	*/
 	public function split_keywords(&$keywords, $terms)
 	{
+		// Keep quotes and new lines
+		$keywords = str_replace(array('&quot;', "\n"), array('"', ' '), trim($keywords));
+
 		if ($terms == 'all')
 		{
-			$match		= array('#\sand\s#i', '#\sor\s#i', '#\snot\s#i', '#\+#', '#-#', '#\|#', '#@#');
-			$replace	= array(' & ', ' | ', '  - ', ' +', ' -', ' |', '');
+			// Replaces verbal operators OR and NOT with special characters | and -, unless appearing within quotation marks
+			$match		= array('#\sor\s(?=([^"]*"[^"]*")*[^"]*$)#i', '#\snot\s(?=([^"]*"[^"]*")*[^"]*$)#i');
+			$replace	= array(' | ', ' -');
 
 			$keywords = preg_replace($match, $replace, $keywords);
 			$this->sphinx->SetMatchMode(SPH_MATCH_EXTENDED);
 		}
 		else
 		{
+			$match = array ( '\\', '(',')','|','!','@','~', '/', '^', '$', '=','&amp;', '&lt;', '&gt;');
```

Comment:
This should be without spacing, `array('\\',` instead of `array ( '\\',`.
And ideally with new array syntax `[]`.